tests/manual/test_lazy_load.py

```diff
@@ -0,0 +1,284 @@
+        self.assertEqual(8, len(all_direct_files))
+
+    def test_get_files_and_folders(self):
+        ds005 = load_dataset(DS005_DIR)
+
+        file = ds005.get_file("dataset_description.json")
+        self.assertIsNotNone(file)
+        self.assertEqual("dataset_description.json", file.name)
+
+        file = ds005.get_file("sub-01/func/sub-01_task-mixedgamblestask_run-01_bold.nii.gz")
+        self.assertIsNotNone(file)
+        self.assertEqual("sub-01_task-mixedgamblestask_run-01_bold.nii.gz", file.name)
+
+    def test_repr(self):
+        ds005 = load_dataset(DS005_DIR)
+        self.assertEqual("{'name': 'ds005'}", str(ds005))
+        self.assertEqual("{'name': 'derivatives'}", str(ds005.derivatives))
+        self.assertEqual("{'name': 'README'}", str(ds005.README))
+        expected = "{'name': 'dataset_description.json', 'Name': 'Mixed-gambles task', 'BIDSVersion': '1.0.0rc2', 'License': 'This dataset is made available u[...]'}"
+        self.assertEqual(expected, str(ds005.dataset_description))
+
+    def test_participants_tsv(self):
+        ds005 = load_dataset(DS005_DIR)
+        schema = ds005.get_schema()
+        self.assertTrue(isinstance(ds005.participants_tsv, schema.TSVFile))
+        contents = ds005.participants_tsv.contents
+        self.assertTrue(contents is not None)
+        self.assertEqual(16, len(contents))
+        self.assertEqual(['participant_id', 'sex', 'age'], list(contents[0].keys()))
+        self.assertEqual(['sub-01', '0', '28'], list(contents[0].values()))
+
+        # or short form:
+        self.assertEqual({'participant_id': 'sub-01', 'sex': '0', 'age': '28'}, contents[0])
+
+    def test_absolute_path(self):
+        ds_path_norm = os.path.normpath(DS005_DIR)
+        ds005 = load_dataset(ds_path_norm)
+        ds_path = ds005.get_absolute_path()
+        self.assertEqual(ds_path, ds_path_norm)
+
+    def test_datatype_of_artifact(self):
+        ds005 = load_dataset(DS005_DIR)
+        anat_files = ds005.query(scope="raw", sub="01", suffix="T1w")
+        assert len(anat_files) == 1
+        assert anat_files[0].datatype is None
+
+        ds005 = load_dataset(DS005_DIR, DatasetOptions(infer_artifact_datatype=True))
+        anat_files = ds005.query(scope="raw", sub="01", suffix="T1w")
+        assert len(anat_files) == 1
+        assert anat_files[0].datatype == "anat"
+
+    # NEW LAZY LOADING TESTS
+    #for each of the file type and functionality
+    def test_lazy_loading_enabled(self):
+        """Test that lazy loading prevents immediate content loading"""
+        # Load dataset with lazy loading enabled
+        options = DatasetOptions(lazy_loading=True)
+        ds005 = load_dataset(DS005_DIR, options)
+
+        # Check that TSV files are lazy-loaded
+        participants_file = ds005.get_file("participants.tsv")
+        self.assertTrue(hasattr(participants_file, 'set_lazy_loading'))
```

Comment:
This is asserting for internal implementation specific details which may break the unit tests if the implementation changes. The unit test should treat the implementation as a black-box and only infer behavior that is observable from outside to be stable over time.

Reply:
Not yet sure what the best assertions are, but maybe calculate the object memory with and without lazy-loading,  as there is no public interface you can test against, only lazy_loading=True/False in the options.

Reply:
You could also use a mocking library to intercept the internal function calls to determine if a load function was called or not (depending on lazy_load flag in options)